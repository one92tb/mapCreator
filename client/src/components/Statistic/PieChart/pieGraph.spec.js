import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import { PieGraph } from "./PieGraph";

describe(" pie graph component", () => {
  let wrapper;

  it("passes all props to pieGraph component", () => {
    const displayMarkers = jest.fn(() => {
      return [["All markers", 2]];
    });

    wrapper = shallow(<PieGraph displayMarkers={displayMarkers} />);
    expect(wrapper).toMatchSnapshot();
  });
});
