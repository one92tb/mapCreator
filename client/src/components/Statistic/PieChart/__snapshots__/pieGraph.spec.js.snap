// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` pie graph component passes all props to pieGraph component 1`] = `
<ResponsiveContainer
  debounce={0}
  height="100%"
  width="100%"
>
  <PieChart
    barCategoryGap="10%"
    barGap={4}
    cx="50%"
    cy="50%"
    endAngle={360}
    innerRadius={0}
    layout="centric"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    outerRadius="80%"
    reverseStackOrder={false}
    stackOffset="none"
    startAngle={0}
  >
    <Pie
      activeIndex={0}
      activeShape={[Function]}
      animationBegin={400}
      animationDuration={1500}
      animationEasing="ease"
      blendStroke={false}
      cx={300}
      cy={170}
      data={
        Array [
          Object {
            "name": "All markers",
            "value": 2,
          },
        ]
      }
      dataKey="value"
      endAngle={360}
      fill="#8884d8"
      hide={false}
      innerRadius={60}
      isAnimationActive={true}
      labelLine={true}
      legendType="rect"
      minAngle={0}
      nameKey="name"
      onMouseEnter={[Function]}
      outerRadius={80}
      paddingAngle={0}
      startAngle={0}
      stroke="#fff"
    />
  </PieChart>
</ResponsiveContainer>
`;
